File: 02_R/01_monitoring_VAP5.R
#' ---
#' project: Zivi2022R    ##################################################
#' title:   Monitoring VAP
#' author:  Reto Zihlmann <retozihlmann@outlook.com>
#' date:    2022-07-05 16:12:43
#' output:  github_document   #############################################
#' ---


# packages ----------------------------------------------------------------

library(readxl)
library(magrittr)
library(tidyverse)
source("02_R/00_fun.R")



# read data ---------------------------------------------------------------

dat <- read_excel_n("01_data/vap_monitoring_output.xlsx")

dat <- dat %>% 
  # recode response variables
  mutate(OKH_Formel = ifelse(OKH_Formel == "korrekt", 1,
                             ifelse(OKH_Formel == "nicht korrekt", 0, NA)),
         Sedationsstopp_Ja_Nein = ifelse(Sedationsstopp_Ja_Nein == "Ja", 1,
                                         ifelse(Sedationsstopp_Ja_Nein == "Nein",
                                                0, NA)),
         Mundpflege = ifelse(Mundpflege == "Ja", 1,
                             ifelse(Mundpflege == "Nein", 0, NA)))


# create tables -----------------------------------------------------------

t_OKH <- prop_period_fun(dat = dat, response = "OKH_Formel",
                         by = "Station",
                         filter_crit = "OKH_istRelevant")
t_Sed <- prop_period_fun(dat = dat, response = "Sedationsstopp_Ja_Nein",
                         by = "Station",
                         filter_crit = "Sedationsstopp_istRelevant")
t_Mund <- prop_period_fun(dat = dat, response = "Mundpflege",
                         by = "Station",
                         filter_crit = "Mundpflege_istRelevant")


# create plots ------------------------------------------------------------

p_OKH <- plot_prop_period(prop_period = t_OKH,
                          fileprefix = "01_VAP_OKH")
p_Sed <- plot_prop_period(prop_period = t_Sed,
                          fileprefix = "01_VAP_Sed")
p_Mund <- plot_prop_period(prop_period = t_Mund,
                           fileprefix = "01_VAP_Mund")

File: 02_python/01_monitoring_VAP5.py
from fun import *

df1 = pd.read_excel("../01_data/VAP_python_output2_cens.xlsx", engine='openpyxl')

df1["OKH_Formel"] = df1["OKH_Formel"].map({'korrekt': 1, 'nicht korrekt': 0})
df1["Sedationsstopp_Ja_Nein"] = df1["Sedationsstopp_Ja_Nein"].map({'Ja': 1, 'Nein': 0})
df1["Mundpflege"] = df1["Mundpflege"].map({'Ja': 1, 'Nein': 0})

t_OKH = prop_period_fun(df1, "OKH_Formel", "Station", "OKH_istRelevant")
t_Sed = prop_period_fun(df1, "Sedationsstopp_Ja_Nein", "Station", "Sedationsstopp_istRelevant")
t_Mund = prop_period_fun(df1, "Mundpflege", "Station", "Mundpflege_istRelevant")

# print(t_OKH)

p_OKH = plot_prop_period(t_OKH, "../03_figures/01_VAP_OKH")
p_Sed = plot_prop_period(t_Sed, "../03_figures/01_VAP_Sed")
p_Mund = plot_prop_period(t_Mund, "../03_figures/01_VAP_Mund")